tests/test.py

```diff
@@ -0,0 +1,106 @@
+        # check title
+        self.assertIn('CodaLab - Home', self.browser.title)
+
+        for item in ['Worksheets', 'Competitions', 'Help', 'Example Worksheets', 'Latest Competitions']:
+            self.assertIn(item, self.browser.page_source)
+
+    def testWorksheetContent(self):
+        cl = 'cl'
+        self.browser.get(self.base_url)
+        self.browser.maximize_window()
+        
+        # create a temp worksheet for testing
+        temp_worksheet_name = 'test-codalab-' + str(random.randint(0, 1000000))
+        temp_worksheet_title =  'The really long testing title for testing the automated testing of worksheets'
+        temp_worksheet_uuid = self.run_command([cl, 'new', temp_worksheet_name])
+        temp_file_name = 'cl-web-interface-testing.txt'
+        temp_worksheet_paragraph_content = "For running automated testing on the cli " * 10
+        self.run_command([cl, 'work', temp_worksheet_uuid])
+        self.run_command([cl, 'wedit', '-t', temp_worksheet_title])
+        self.run_command([cl, 'add',  '-m', temp_worksheet_paragraph_content])
+        self.run_command(['touch', temp_file_name])
+        temp_file_uuid = self.run_command([cl, 'upload', 'dataset', temp_file_name])
+
+        # click the worksheet button on the top right of the main page
+        self.browser.find_element_by_xpath("html/body/nav/div/div[2]/ul/li[1]/a").click()
+
+        # find the worksheet that was just created
+        self.browser.implicitly_wait(4)
+        self.browser.get(self.browser.find_element_by_link_text(temp_worksheet_name).get_attribute('href'))
+
+        # wait for a sec to make sure that the AJAX calls have been completed.
+        # TODO: find a better and more selenium way to do it.
+        import time
+        time.sleep(1)
+
+        # Make sure that there is no error displayed on the page
+        self.assertNotIn('ERROR', self.browser.page_source)
+
+        # check worksheet layout
+        self.worksheetLayoutCheckHelper(temp_worksheet_name, temp_worksheet_uuid, temp_worksheet_title)
+
+        # check worksheet sidepanel
+        self.worksheetSidePanelCheckHelper(temp_worksheet_name, temp_worksheet_uuid, temp_file_name)
+
+        # check worksheet content
+        self.worksheetContentCheckHelper(temp_file_name, temp_file_uuid, temp_worksheet_paragraph_content)
+
+        # delete the temp-worksheet
+        self.run_command([cl, 'wrm', '--force', temp_worksheet_name])
+
+        self.run_command(['rm', temp_file_name])
+
+    def worksheetContentCheckHelper(self, temp_file_name, temp_worksheet_uuid, temp_worksheet_paragraph_content):
+        # test paragraph content inserted at top of temp_worksheet
+        self.assertIn(temp_worksheet_paragraph_content, self.browser.page_source)
+        # check table in worksheet
+        table_content = ['uuid','name', temp_file_name, 'description', 'bundle_type', 'dataset', 'created', 'dependencies', 'command', 'MISSING', 'data_size', '0','state', 'ready']
```

Comment:
oh no. I know what is going on. The branch had not been updated and it still shows MISSING. I did not realize that. I will pull the latest code and revise this. Sorry my bad.